kin-devplatform-sdk/src/main/java/kin/devplatform/KinEcosystemInitiator.java

```diff
@@ -76,85 +80,109 @@ public void internalInit(Context context) {
+			}
+		}
+	}
 
-			EventCommonDataUtil.setBaseData(context);
+	private void handleMigration(final Context context, final String appId, final EventLogger eventLogger, MigrationManager migrationManager,
+								 final boolean withLogin, final SignInData signInData, final KinCallback<Void> loginCallback) throws MigrationInProcessException {
+		migrationManager.start(new MigrationManagerListener() {
+			@Override
+			public void onMigrationStart() {
+				// TODO: 06/01/2019 when implement the features we talked about with Ayelet then add here the onMigrationStarted for the external developers to use.
+			}
 
-			AccountManagerImpl
-				.init(AccountManagerLocal.getInstance(context), eventLogger, AuthRepository.getInstance(),
-					BlockchainSourceImpl.getInstance());
+			@Override
+			public void onReady(IKinClient kinClient) {
+				try {
+					String applicationId = appId;
+					BlockchainSourceImpl.init(eventLogger, kinClient, BlockchainSourceLocal.getInstance(context));
+					if (applicationId != null) {
+						BlockchainSourceImpl.getInstance().setAppID(applicationId);
+					}
 
-			OrderRepository.init(BlockchainSourceImpl.getInstance(),
-				eventLogger,
-				OrderRemoteData.getInstance(executorsUtil),
-				OrderLocalData.getInstance(context, executorsUtil));
+					applicationId = BlockchainSourceImpl.getInstance().getAppId();
 
-			OfferRepository
-				.init(OfferRemoteData.getInstance(executorsUtil), OrderRepository.getInstance());
+					AuthRepository
+							.init(AuthLocalData.getInstance(context, executorsUtil),
+									AuthRemoteData.getInstance(executorsUtil), applicationId);
 
-			DeviceUtils.init(context);
+					EventCommonDataUtil.setBaseData(context);
 
-			setAppID();
+					AccountManagerImpl
+							.init(AccountManagerLocal.getInstance(context), eventLogger, AuthRepository.getInstance(),
+									BlockchainSourceImpl.getInstance());
 
-			eventLogger.send(KinSdkInitiated.create());
-			isInitialized = true;
-		}
-	}
+					OrderRepository.init(BlockchainSourceImpl.getInstance(),
+							eventLogger,
+							OrderRemoteData.getInstance(executorsUtil),
+							OrderLocalData.getInstance(context, executorsUtil));
 
-	private void setAppID() {
-		ObservableData<String> observableData = AuthRepository.getInstance().getAppID();
-		String appID = observableData.getValue();
-		if (appID == null) {
-			observableData.addObserver(new Observer<String>() {
-				@Override
-				public void onChanged(String appID) {
-					if (!TextUtils.isEmpty(appID)) {
-						BlockchainSourceImpl.getInstance().setAppID(appID);
-						AuthRepository.getInstance().getAppID().removeObserver(this);
+					OfferRepository
+							.init(OfferRemoteData.getInstance(executorsUtil), OrderRepository.getInstance());
+
+					DeviceUtils.init(context);
+
+					eventLogger.send(KinSdkInitiated.create());
+					isInitialized = true;
+					if (withLogin) {
+						login(signInData, loginCallback);
 					}
+				} catch (BlockchainException e) {
+					onError(e);
```

Comment:
should be handled separately from migration `onError`, probably via `fireStartError` but only if `withLogin` is enable, o.w. should rethrow

Reply:
Fix it, now if with login the fireStartError like it was done before in externalInit method.
Else then just log the error like it was done before in internalInit method